doc/proposals/operator-testing-tool.md

```diff
@@ -0,0 +1,226 @@
+---
+
+# operator-testing-tool
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This proposal is for a new Operator Testing Tool that validates Operator packaging as well as run simple functional test specific to the Operator’s APIs.
+
+## Motivation
+
+Operator Testing and Validation remains difficult today due to tooling sprawl, disparate definitions of validity, and the lack of means to provide custom tests per Operator in a way that is not tied to a particular CI pipeline.
+
+Having high quality Operators is crucial to the success of the Operator Framework.
+
+### Goals
+
+ * Operator Developers can use a tool that reports on recommended, required and optional packaging requirements of a given bundle on disk
+ * Operator Developers can use the same tool that reports on recommended, required and optional common black box tests of a deployed operator on cluster
+ * Operator Developers can rely on a central source of truth of a valid Operator bundles shared among all Operator Framework components
+
+
+### Non-Goals
+
+not in scope: Operator Developers can use the same tool to run custom, functional tests of their Operator on cluster
+
+## Proposal
+
+Details for this proposal are define in the following user stories.
+
+### User Stories
+
+#### Story 1 - Show pass/fail in Scorecard Output
+
+Today, the scorecard output shows a percentage of tests
+that were successful back to the end user.  This story is to
+change the scorecard output to show a *pass* or *fail* for each
+test that is run in the output instead of a success percentage.  The
+exit code of scorecard would be 0 if all tests passed.  The exit
+code would be non-zero if tests failed.  With this change
+scores are essentially replaced with a list of pass/fail(s).
+
+#### Story 2 - Change Scorecard Test Selection
+
+Today, the scorecard lets you select which tests are run by
+the plugins defined in the scorecard configuration (basic, olm).  This story
+would change how users determine which tests are executed by scorecard.
+
+This story would introduce labels for each test and then allow the
+scorecard CLI to filter which tests it runs based on label selectors.
+
+Tests can fall into 2 groups: *required*, or *optional*.  Tests
+can also be categorized as *static* or *runtime*.  Labels for
+these groups and categories would allow a test to be more precisely
+specified by an end user.
+
+Possible examples of specifying what tests to run are as follows:
+ * operator-sdk scorecard --selector="required,runtime"
+ * operator-sdk scorecard --selector="required,runtime,static"
+ * operator-sdk scorecard --selector="optional,static"
+
+A scheme for applying labels to the actual tests would need to
+be developed.
```

Comment:
Is there a way to run tests by name? I know when I run `go test` I often use the run command to run a single test.